pkg/common/cluster_actions.go

```diff
@@ -94,10 +131,44 @@ type GenericUpdateAction struct {
 	Msg string
 }
 
+type CreateRealmAction struct {
+	Ref *v1alpha1.KeycloakRealm
+	Msg string
+}
+
+type DeleteRealmAction struct {
+	Ref *v1alpha1.KeycloakRealm
+	Msg string
+}
+
+type NoopAction struct {
```

Comment:
I think this just adds complexity and doesn't buy us anything. IMO we should avoid adding a new action if there's nothing to do.

Reply:
I added this because currently we expect every `getDesiredStateForX` function is expected to return an action. Sometimes there is nothing to do. But i'd agree that we should rather change this convention than introduce an action that does nothing.

Reply:
We might return a pointer to an `Action`. I think that's how it was implemented initially. I don't know if there's anything like `Optional` in Go.